### PR TITLE
287_find_duplicate_number

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code

### CODE REVIEW

File: 287_find_duplicate_number.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import List
+
+
+class Solution:
+    def findDuplicate(self, nums: List[int]) -> int:
+        nums.sort()
+        for i in range(len(nums)):
+            if nums[i] == nums[i + 1]:
+                return nums[i]
```
Comment (ryujumoto) [inline 287_find_duplicate_number.py]: Use binary search.

File: 287_find_duplicate_number.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from typing import List
```
Comment (ryujumoto) [inline 287_find_duplicate_number.py:3]: Write time complexity and space complexity.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I revised it. Please check

File: 287_find_duplicate_number.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# time complexity O(logn)
+# space complexity O(1)
```
Comment (ryujumoto) [inline 287_find_duplicate_number.py:2]: Space complexity is wrong.
Investigate space complexity of Python's sort method.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I learned that the .sort() requires O(nlogn)
Please check my code.

File: 287_find_duplicate_number.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+from typing import List
+
+
+class Solution:
+    def findDuplicate(self, nums: List[int]) -> int:
+        n = len(nums) - 1
+        left, right = 1, n
+        while left < right:
+            mid = (right + left) // 2
```
Comment (ryujumoto) [inline 287_find_duplicate_number.py]: mid = (right+left) // 2

**Comment by Rwatana** [comment]
I revised it. Please check my code.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**